addons/project/static/src/components/project_task_state_selection/project_task_stage_state_selection/project_task_stage_with_state_selection.js

```diff
@@ -0,0 +1,55 @@
+import { _t } from "@web/core/l10n/translation";
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { Many2OneField } from "@web/views/fields/many2one/many2one_field";
+
+import { registry } from "@web/core/registry";
+import { Component } from "@odoo/owl";
+
+import { ProjectTaskStateSelection } from "../project_task_state_selection";
+
+
+export class TaskStageWithStateSelection extends Component {
+    static template = "project.TaskStageWithStateSelection";
+
+    static props = {
+        ...standardFieldProps,
+        stateReadonly: { type: Boolean, optional: true },
+        viewType: { type: String },
+    };
+
+    static components = {
+        ProjectTaskStateSelection,
+        Many2OneField,
+    };
+
+    setup() {
+        this.M2Oprops = { ...(({ stateReadonly, viewType, ...o }) => o)(this.props) };
+        this.stateProps = {
+            ...this.M2Oprops,
+            name: "state",
+            readonly: this.props.stateReadonly,
+            viewType: this.props.viewType,
+            showLabel: false,
+        };
+    }
+}
```

Comment:
I would personally create a getter for `stateProps` and `stageProps`. 🤷🏼‍♂️ 

```suggestion
    get stageProps() {
        return omit(this.props, "stateReadonly", "viewType");
    }
    
    get stateProps () {
        return {
            ...this.stateProps,
            name: "state",
            readonly: this.props.stateReadonly,
            viewType: this.props.viewType,
            showLabel: false,
        };
    }
}
```

What do you think? 🙂

Reply:
:+1: It looks good and this was my very first approach as well. But I don't seem to understand why this works here at first ```stageProps``` is accessed and we omit(returns a shallow copy by removing mentioned key-pair values). Shouldn't the changes reflect in ```this.props``` as it was done on shallow copy. So In theory ```stateProps``` should fail to get ```this.props.stateReadonly``` and ```this.props.viewType```. Well after checking the omit function is returning a new object not a shallow copy :cry: . The documentation need to be changed(I was fooled by the documentation of omit method). It seems clear now. Thanks for the suggestion

Reply:
![minion-memory-alert](https://github.com/user-attachments/assets/e0b44ea3-0588-4ade-99ad-0ef8e4b43825)

Calling stateProps from the stateProps it self :smirk: